src/ray/common/ray_config_def.h

```diff
@@ -661,16 +661,22 @@ RAY_CONFIG(bool, isolate_workers_across_task_types, true)
 /// ServerCall instance number of each RPC service handler
 RAY_CONFIG(int64_t, gcs_max_active_rpcs_per_handler, 100)
 
-/// grpc keepalive sent interval
+/// grpc keepalive sent interval for server.
 /// This is only configured in GCS server now.
-/// NOTE: It is not ideal for other components because
+RAY_CONFIG(int64_t, grpc_keepalive_time_ms, 10000)
+
+/// grpc keepalive timeout.
+RAY_CONFIG(int64_t, grpc_keepalive_timeout_ms, 20000)
+
+/// NOTE: we set a loose client keep alive because
 /// they have a failure model that considers network failures as component failures
 /// and this configuration break that assumption. We should apply to every other component
 /// after we change this failure assumption from code.
-RAY_CONFIG(int64_t, grpc_keepalive_time_ms, 10000)
+/// grpc keepalive timeout for client.
+RAY_CONFIG(int64_t, grpc_client_keepalive_time_ms, 60000)
 
-/// grpc keepalive timeout
-RAY_CONFIG(int64_t, grpc_keepalive_timeout_ms, 20000)
+/// grpc keepalive timeout for client.
+RAY_CONFIG(int64_t, grpc_client_keepalive_timeout_ms, 120000)
```

Comment:
Would this result in `GOAWAY` warnings? 

From [GRPC](https://github.com/grpc/grpc/blob/master/doc/keepalive.md)

> Why am I receiving a GOAWAY with error code ENHANCE_YOUR_CALM?
> - if the client's GRPC_ARG_KEEPALIVE_TIME_MS setting is lower than the server's GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS.

So IIUC, with the default `GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS=300000` on server  and `GRPC_ARG_HTTP2_MAX_PING_STRIKES=2` by default, the server would start sending GOAWAY at the 3rd ping right?

Reply:
Yeah I think we should either increase the ping frequency to 5 minutes or reduce `GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS ` to 1 minute

Reply:
i can't do math or understand this sophisticated grpc config. but seems we set it default to 1 minutes https://sourcegraph.com/github.com/ray-project/ray/-/blob/src/ray/rpc/grpc_server.cc?L93 
should i change my value?

Reply:
Ah in this case, this should be okay.

Reply:
we need to make sure this is set to 1 minute in python grpc servers too though.. otherwise it can create GOAWAY warning